lua/pac3/core/client/parts/event.lua

```diff
@@ -1424,11 +1439,9 @@ function PART:GetParsedArguments(data)
 	local args = line:Split("@@")
 
 	for pos, arg in pairs(data) do
-		local typ = select(2, next(arg))
+		local typ = arg[2]--select(2, next(arg))

```

Comment:
typ is supposed to be the argument's type, which is at arg[2] (I haven't seen an instance where it isn't arg[2]).
select(2, next(arg)) was returning the argument's name only which meant the type wasn't being properly retrieved. If there's a better way of retrieving the type then feel free to change it of course.

Reply:
Yeah, if the previous code was wrong then that's good. I'd remove the comment though.

Reply:
I could do `local typ = arg[2] or select(2, next(arg))` so if the type isn't found in the usual spot, it will default back to what it looked at previously - just incase?

Reply:
I don't think so. It should be consistent in my opinion, but I don't know how it works or worked so I'll let the others decide.

Reply:
Looks like the code that builds the table 'data' is defined here

[`table.insert(self.__registeredArguments, {keyName, keyType, userdata})`](https://github.com/CapsAdmin/pac3/blob/eb85fef769b2ea1301d74596525a287823f6ba37/lua/pac3/core/client/parts/event.lua#L1040)

This is inside the function "AppendArguement" which has multiple checks to ensure the second parameter, keyType, is valid. However has no checks to make sure that keyName is valid, which means it could be nil. 

In this case, the table to be added to data would be {nil, keyType, nil}, which would make data look like this:

Print table output:
> 1:
		2	=	boolean


Because of this, it's possible that the variable 'arg' could have no value at arg[1], however still have a value at arg[2].
Hence, using arg[2] would be safer in my opinion.

Reply:
Yeah, I agree